includes/admin.inc

```diff
@@ -1278,6 +1359,16 @@ function _islandora_solr_admin_settings_validate($form, &$form_state) {
   $limits = explode(',', $form_state['values']['islandora_solr_num_of_results_advanced']);
   $form_state['values']['islandora_solr_num_of_results_advanced'] = array_filter(array_map('trim', $limits), 'is_numeric');
   $form_state['values']['islandora_solr_namespace_restriction'] = preg_replace('/:$/', '', $form_state['values']['islandora_solr_namespace_restriction']);
+
+  if (preg_match('/^\d*$/', $form_state['values']['islandora_solr_query_phrase_slop']) !== 1) {
+    form_set_error('islandora_solr_query_phrase_slop', t("Query phrase slop should be a number."));
+  }
+  if (preg_match('/^\d*$/', $form_state['values']['islandora_solr_phrase_slop']) !== 1) {
+    form_set_error('islandora_solr_phrase_slop', t("Phrase slop should be a number."));
+  }
+  if (preg_match('/^(?:0\.\d+|1.0|)$/', $form_state['values']['islandora_solr_tie_breaker']) !== 1) {
```

Comment:
Same, although by now I'm doubting whether you should check numeric values using regular expressions at all.

Reply:
I use regular expressions because I want more than a check on numeric values: this one checks if the number is a float between 0.0 and 1.0 or is empty.